### PR TITLE
Revamped Chains form

### PR DESCRIPTION
**Pull request by vyzaldysanchez**
Closes: https://app.shortcut.com/chainlinklabs/story/18604/ui-chain-specific-configuration-is-clunky-unintuitive

This aims to remove the need to paste JSON for the chain configuration that it is not `Key Specific`, enabling text inputs and dropdowns.

UI Preview:

gnome-shell-screenshot-PCC3B1
gnome-shell-screenshot-ZFG2B1

### CODE REVIEW

**Comment by vyzaldysanchez** [comment]
@jkongie any words on how testing for this type of UI components should happen or be written? I looked at the `FeedsManagerForm` component for reference, and that one does not have tests either.

**Comment by jkongie** [comment]
> @jkongie any words on how testing for this type of UI components should happen or be written? I looked at the `FeedsManagerForm` component for reference, and that one does not have tests either.

As we are transitioning over to the GQL API we will also be starting to use a new testing library. Verify that it works manually for now, and we will add tests in during the refactor. I've already started it but just need a couple of my PRs to get in before I can complete it

**Review by samsondav — DISMISSED** [review]
Nice!

**Review by samsondav — COMMENTED** [review]
Does it also allow to update existing chains?

**Comment by vyzaldysanchez** [comment]
@samsondav yes, it allows to modify existing chains.

**Review by samsondav — APPROVED** [review]

**Review by samsondav — APPROVED** [review]

File: operator_ui/src/pages/Chains/ChainConfigFields.tsx [hunk 1]
```diff
@@ -0,0 +1,628 @@
...
+export interface ConfigOverrides {
+  [attr: string]: JSONPrimitive
+}
+
+type NonNullableJSONPrimitive = string | number | boolean
+
+const defaultKeySpecifics = '{}'
+
+const chainTypes = ['arbitrum', 'exchain', 'optimism', 'xdai']
+const gasEstimatorModes = ['BlockHistory', 'FixedPrice', 'Optimism']
```
Comment (jmank88) [inline operator_ui/src/pages/Chains/ChainConfigFields.tsx:27]: There is also an `Optimism2` gas estimator.
```suggestion
const gasEstimatorModes = ['BlockHistory', 'FixedPrice', 'Optimism', 'Optimism2']
```

**Review by samsondav — APPROVED** [review]

**Review by jmank88 — APPROVED** [review]

**Pull request merged**